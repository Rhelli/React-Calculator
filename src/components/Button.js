import React from 'react';
import PropTypes from 'prop-types';

const Button = props => {
  const { buttonName } = props;

  const handleClick = event => {
    event.stopPropagation();
    props.clickHandler(event.target.id);
  };

  return (
    <button
      type="button"
      id={buttonName}
      onClick={handleClick}
    >
      {buttonName}
    </button>
  );
};

Button.defaultProps = {
  buttonName: '',
  clickHandler: null,
};

Button.propTypes = {
  buttonName: PropTypes.string,
  clickHandler: PropTypes.func,
};

export default Button;
