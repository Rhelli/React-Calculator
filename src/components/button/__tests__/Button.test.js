import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('Button appearance', () => {
  it('Matches the Button Snapshot', () => {
    const tree = renderer.create(<Button />).toJSON()
    expect(tree).toMatchSnapshot()
  });

  it('Matches its inline snapshot', () => {
    const { container } = render(<Button />);

    expect(container.firstChild).toMatchInlineSnapshot(`
      <div
        class="button"
      >
        <button
          class="grey"
          id=""
          type="button"
        />
      </div>
    `)
  });
});
