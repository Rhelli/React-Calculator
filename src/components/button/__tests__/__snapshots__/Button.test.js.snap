// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button appearance Matches the Button Snapshot 1`] = `
<div
  className="button"
>
  <button
    className="grey"
    id=""
    onClick={[Function]}
    type="button"
  >
    
  </button>
</div>
`;
